operator/postgresqlcontroller/observe.go

```diff
@@ -64,3 +105,81 @@ func setConditionFromState(pgExo oapi.DbaasServicePg, pgInstance *exoscalev1.Pos
 		pgInstance.SetConditions(exoscalev1.Rebalancing())
 	}
 }
+
+var variantAiven = oapi.EnumPgVariantAiven
+
+// mapObservation fills the status fields from the given response body.
+func mapObservation(pg oapi.DbaasServicePg) (exoscalev1.PostgreSQLObservation, error) {
+	observation := exoscalev1.PostgreSQLObservation{
+		DBaaSParameters: exoscalev1.DBaaSParameters{
+			TerminationProtection: pointer.BoolDeref(pg.TerminationProtection, false),
+			Size: exoscalev1.SizeSpec{
+				Plan: pg.Plan,
+			},
+			IPFilter: *pg.IpFilter,
+		},
+		Version: pointer.StringDeref(pg.Version, ""),
+		Maintenance: exoscalev1.MaintenanceSpec{
+			DayOfWeek: pg.Maintenance.Dow,
+			TimeOfDay: exoscalev1.TimeOfDay(pg.Maintenance.Time),
+		},
+		Backup:     mapper.ToBackupSpec(pg.BackupSchedule),
+		NodeStates: mapper.ToNodeStates(pg.NodeStates),
+	}
+
+	settings, err := mapper.ToRawExtension(pg.PgSettings)
+	if err != nil {
+		return observation, errors.Wrap(err, "cannot marshal json")
+	}
+	observation.PGSettings = settings
+
+	return observation, nil
+}
+
+// isUpToDate returns true if the observed response body matches the desired spec.
+func isUpToDate(current, external *exoscalev1.PostgreSQLParameters, log logr.Logger) bool {
+	if external == nil {
+		return false
+	}
+	sameMajorVersion, err := mapper.CompareMajorVersion(current.Version, external.Version)
+	if err != nil {
+		log.Error(err, "parse PostgreSQL version", "current", current.Version, "external", external.Version)
+	}
+	extIPFilter := []string(external.IPFilter)
+	checks := map[string]bool{
+		"IPFilter":              mapper.IsSameStringSet(current.IPFilter, &extIPFilter),
+		"MajorVersion":          sameMajorVersion,
+		"Maintenance":           current.Maintenance.Equals(external.Maintenance),
+		"BackupSchedule":        current.Backup.Equals(external.Backup),
+		"Size":                  current.Size.Equals(external.Size),
+		"TerminationProtection": current.TerminationProtection == external.TerminationProtection,
+		"PGSettings":            mapper.CompareSettings(current.PGSettings, external.PGSettings),
+	}
+	ok := true
+	for _, v := range checks {
+		if !v {
+			log.V(2).Info("instance not up-to-date", "check", v)
+			ok = false
+		}
+	}
+	return ok
+}
+
+// toConnectionDetails parses the connection details from the given observation.
+func toConnectionDetails(pgExo oapi.DbaasServicePg, ca string) (managed.ConnectionDetails, error) {
```

Comment:
nit: maybe rename it to `connectionDetails` to be consistent with other resource's observe methods